### ISSUE TITLE
Broken Database Seed

### ISSUE THREAD

**Issue by brandonjank** [uidaho/squireproject#79]
I enabled seeding on travis-ci builds and one of the seeds is broken:

```
The command "php artisan migrate" exited with 0.
0.34s$ php artisan db:seed
Seeded: UsersTableSeeder

  [ErrorException]     
  Undefined offset: 1  

The command "php artisan db:seed" exited with 1.
```

**Comment by brandonjank** [comment]
keep in mind that travis-ci makes a fresh install every time, you will not be able to reproduce this bug without an empty and then freshly migrated database

**Comment by brec9824** [comment]
Yeah I just noticed this. I think have it fixed but I am doing some more testing.

**Comment by brandonjank** [comment]
in ModelFactory.php:

```
//Creates projects and uses previously made users
$factory->define(Project::class, function (Faker\Generator $faker) {
    $randUserNum = random_int(1, DB::table('users')->count());
    $username = DB::table('users')->lists('username')[$randUserNum];

    return [
        'title' => $faker->catchPhrase,
        'author' => $username,
        'description' => $faker->bs,
        'body' => $faker->paragraph,
    ];
});
```

this doesnt work, the factory should create users it needs. DB::table('users')->count() returns 0 and you cant generate a number between 1 and 0  other than 1. and there are no users in the table thus the index error.

**Comment by brandonjank** [comment]
tldr; you'll need to empty your users table to test if the seed is working

**Comment by brandonjank** [comment]
rewrote it:

```
// Creates projects using random users from the users table
$factory->define(Project::class, function (Faker\Generator $faker) {
    $user = User::all()->random(1);

    return [
        'title' => $faker->catchPhrase,
        'author' => $user->username,
        'description' => $faker->bs,
        'body' => $faker->paragraph,
    ];
});
```

the test should call the user factory and then the project factory

factory created items are purged between tests

**Comment by brec9824** [comment]
Yeah thats what I thought

**Comment by brandonjank** [comment]
also,  $faker providers should be using unique()  for titles, usernames, and e-mails. We dont want duplicate keys causing false positives.

e.g.
`'title' => $faker->catchPhrase,`

should be:
`'title' => $faker->unique()->catchPhrase,`

**Comment by brec9824** [comment]
Yeah I made some changes to that. I also realized usernames werent being created with a correct regex
